### PR TITLE
Azure blob storage sas token

### PR DESCRIPTION
**Pull request by SmitaRKulkarni**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Updated check for connection_string as connection string with sas does not always begin with DefaultEndPoint and updated connection url to include sas token after adding container to url.

cc @alesapin

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
This is an automated comment for commit 86d0e542ea9d966b164f90f35ec95acf2579f5b7 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51141/86d0e542ea9d966b164f90f35ec95acf2579f5b7/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by alesapin — CHANGES_REQUESTED** [review]
@SmitaRKulkarni All azure tests failed.

File: src/Storages/StorageAzureBlob.cpp [hunk 1]
```diff
@@ -315,58 +315,90 @@ AzureClientPtr StorageAzureBlob::createClient(StorageAzureBlob::Configuration co
...
+                break;
+            }
+        }
+
         result = std::make_unique<BlobContainerClient>(BlobContainerClient::CreateFromConnectionString(configuration.connection_url, configuration.container));
-        result->CreateIfNotExists();
+
+        if (!container_exists)
+        {
+            result->CreateIfNotExists();
```
Comment (alesapin) [inline src/Storages/StorageAzureBlob.cpp:341]: It it's readonly case maybe it make sense to throw exception?

File: src/Storages/StorageAzureBlob.cpp [hunk 2]
```diff
@@ -315,58 +315,90 @@ AzureClientPtr StorageAzureBlob::createClient(StorageAzureBlob::Configuration co
...
         }
         else
         {
-            auto managed_identity_credential = std::make_shared<Azure::Identity::ManagedIdentityCredential>();
-            auto blob_service_client = std::make_unique<BlobServiceClient>(configuration.connection_url, managed_identity_credential);
-            try
+            blob_service_client = std::make_unique<BlobServiceClient>(configuration.connection_url);
+        }
+
+        Azure::Storage::Blobs::ListBlobContainersOptions options;
```
Comment (alesapin) [inline src/Storages/StorageAzureBlob.cpp:363]: let's move it to a function?

**Review by alesapin — APPROVED** [review]
Test failure looks related

File: src/Storages/StorageAzureBlob.cpp (hunk 1 above)
Comment (SmitaRKulkarni) [reply src/Storages/StorageAzureBlob.cpp:347]: Updated to pass parameter `is_insert_query` to table function (had to update all table functions, but this parameter is currently only used in azureBlobStrorage). If its not insert query and container does not exist, we throw error.

**Review by alesapin — APPROVED** [review]

**Pull request merged**